server/controllers/authController.ts

```diff
@@ -0,0 +1,69 @@
+import { Request, Response } from 'express';
+import { AuthenticatedRequest } from 'express-jwt';
+import { constants } from 'http2';
+import jwt from 'jsonwebtoken';
+import AuthenticationFailedError from '../../shared/errors/authenticationFailedError';
+import { AuthRedirectUrl } from '../../shared/schema/Auth/AuthRedirectUrl';
+import { TokenPayload } from '../../shared/schema/User/TokenPayload';
+import userRepository from '../repositories/userRepository';
+import { getAuthService } from '../services/authService';
+import config from '../utils/config';
+
+const getAuthRedirectUrl = async (request: Request, response: Response) => {
+  const authService = await getAuthService;
+  const authRedirectUrl = authService.getAuthorizationUrl(
+    'openid profile email'
+  );
+
+  response.status(200).send(authRedirectUrl);
+};
+
+const authenticate = async (request: Request, response: Response) => {
+  const authRedirectUrl = request.body as AuthRedirectUrl;
+  const authService = await getAuthService;
+
+  console.log('authenticate', authRedirectUrl);
+
+  try {
+    const { idToken, email } = await authService.authenticate(authRedirectUrl);
+
+    const user = await userRepository.findOne(email);
+    if (!user) {
+      console.error('User not found', email);
+      throw new Error('User not found');
+    }
+
+    const accessToken = jwt.sign(
+      {
+        userId: user.id,
+        idToken
+      } as TokenPayload,
+      config.auth.secret,
+      { expiresIn: config.auth.expiresIn }
+    );
+
+    return response.status(constants.HTTP_STATUS_OK).json({
+      userId: user.id,
+      accessToken,
+      userRoles: user.roles
+    });
+  } catch (error) {
+    console.error('Error while authenticating', error);
+    throw new AuthenticationFailedError();
+  }
+};
+
+const logout = async (request: Request, response: Response) => {
+  const { idToken } = (request as AuthenticatedRequest).auth;
+  const authService = await getAuthService;
+
+  const logoutUrl = authService.getLogoutUrl(idToken);
+
+  return response.status(constants.HTTP_STATUS_OK).json(logoutUrl);
+};
+
+export default {
```

Comment:
C'est vraiment pas pratique le `export default` surtout pour du multiple export.
 - Pour supprimer/ajouter une méthode, faut le modifier la méthode ET le bas du fichier
 - Quand je suis sur une méthode, je ne peux pas savoir si elle est exporter ou pas. Je dois aller tout en bas du fichier pour le savoir
 
Si tu cherches sur le net, la recommandation actuelle est d'éviter les `export default` si c'est possible.

Exemple => 
Using default exports can make it harder to find things in your code and lead to inconsistencies. Instead, named exports make it easier to discover which members are being exported and what their corresponding names are, especially when using an IDE.

It's recommended to avoid using default exports and instead opt for named exports whenever possible. By doing so, you'll have a more organized and maintainable codebase that's easier to work with in the long run.

C'est pas le but de la PR, donc ne change pas. Mais sur les futurs devs, je pense qu'on doit de mettre à utiliser `export const myFunction` partout. Je suis bien entendu dispo pour en parler :smiley: